### ISSUE TITLE
concurrent-output integration?

### ISSUE THREAD

**Issue by mitchellwrosen** [Gabriel439/Haskell-Turtle-Library#123]
Not sure if this is within the scope of what Turtle aims to be, but there's a cool new library by Joey Hess called [concurrent-output](https://hackage.haskell.org/package/concurrent-output) that you might want to look into. 

Even ignoring the awesome console regions API, you could get a lot of mileage out of using `outputConcurrent`/`errorConcurrent` inside of `echo`/`err`, and Turtle users would get de-garbled output from multiple threads for "free". 

Annoyingly, this does require `main = withConcurrentOutput (...)`.

**Comment by Gabriel439** [comment]
I guess my main question here is what is the benefit of `turtle`-specific integration?

**Comment by mitchellwrosen** [comment]
Well, turtle already has some concurrency primitives exposed - fork and wait. However, if two processes are printing to stdout at the same time, their output may be garbled.

For example, if thing1 and thing2 can be run in parallel, but you _also_ want to see the (interleaved) output for debugging or whatever, you can't just get away with

```
a1 <- using (fork thing1)
a2 <- using (fork thing2)
wait a1
wait a2

**Comment by Gabriel439** [comment]
However, the user could get this functionality by just importing the `concurrent-output` package and using it with `turtle`

**Comment by mitchellwrosen** [comment]
Absolutely, but then their scripts have `outputConcurrent` rather than `echo`, and they have to know about the `concurrent-output` package in the first place. I just thought it might be a nice default for `echo` and friends, but as you are pointing out it's probably out of scope. Feel free to close =)

**Comment by Gabriel439** [comment]
Yeah, I think this is probably not worth the extra dependency so I will close this.  Thank you for the suggestion, though
